### ISSUE TITLE
IfcPlant enumeration

### ISSUE THREAD

**Issue by Cafougnette** [bSI-InfraRoom/IFC-Specification#222]
In IfcPlant, the type of PredefinedType is : IfcGeographicElementTypeEnum (deprecated, and not relevant for plants).

Why not create a new enumeration IfcPlantTypeEnum ?
(ENUM_TREE, ENUM_SHRUB, ENUM_HERB, ENUM_GRASS, ENUM_FERN, ENUM_MOSS, ENUM_USERDEFINED, USER_NOTDEFINED)

**Comment by SergejMuhic** [comment]
I am not sure how to handle this properly. This concept is kind of completely off compared to how the schema is structured. `PredefinedType` is on leafs.

I would preferably re-do the concept and deprecate `PredefinedType` from `IfcGeographicElement`, then put it on IfcPlant and introduce the `IfcPlantTypeEnum` that you proposed.

@TLiebich what do you think?

**Comment by TLiebich** [comment]
agree, this is not according to the general IFC data structure, where only leaf-note classes have predefined types. 

There are only some issues around backward compatibility, when deprecating `PredefinedType `at `IfcGeographicElement` we can't introduce the same attribute at a subtype - it would be an override which would lead to syntactic errors.

Is `IfcPlant `- as a sub for landscape architecture - already fully defined for IFC4.3, or can't the proposed addition of enumerators, TREE, SHRUB, HERB, GRASS, FERN, MOSS, PLANT (the last vor any plants) be added to `IfcGeographicElementTypeEnum` and `IfcPlant `be removed?

Or remove PredefinedType from IfcGeographicElement (without deprecation first), it would lead to removing the last attribute in order which most compilers would accept with a warning, and have IfcPlant.PredefinedType : IfcPlantTypeEnum. It would cause less compability issues.

**Comment by SergejMuhic** [comment]
On `IfcPlant` I do not have the background. I don't know where and how the requirement made it into the schema. Both options seem valid but it would be good to know whether the schema is supposed to be used for landscape exchanges. In that case it would make sense to make `IfcGeographicElement` a tree, otherwise a `PredefinedType` would be sufficient.
